src/printer.js

```diff
@@ -2843,7 +2843,11 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
     // the other ones since we don't call the normal print on BinaryExpression,
     // only for the left and right parts
     if (isNested && node.comments) {
-      parts.push(comments.printComments(path, p => "", options));
+      parts.splice(
+        0,
+        parts.length,
+        comments.printComments(path, p => concat(parts.slice()), options)
+      );
```

Comment:
Oh yeah, it's not easy to refactor (I would have done it already :P).

If we've got to chose between two evils, I think that this PR is better. It's not going to indent but at least the comments are going to be stable and at the right place (instead of the crazy way they are now).

Also, this is a very very edge case.